File: api/src/main/java/mock/backend/api/services/user/UserService.java
package mock.backend.api.services.user;


import mock.backend.api.model.user.User;
import mock.backend.api.services.BaseService;

/**
 * Created by semihunaldi on 09.09.2016.
 */
public interface UserService extends BaseService
{
    String testUser();

    User findUserByEmail(String email);

    User findUserById(String id);

    User findUserByUserName(String userName);

    User saveUser(User user);
}

File: api/src/main/kotlin/mock/backend/api/services/user/UserService.kt
package mock.backend.api.services.user

import com.semihunaldi.kotlin.springbootkotlintest.services.BaseService
import mock.backend.api.model.user.User


interface UserService : BaseService
{
    abstract fun testUser(): String?

    abstract fun findUserByEmail(email: String): User?

    abstract fun findUserById(id: String): User?

    abstract fun findUserByUserName(userName: String): User?

    abstract fun saveUser(user: User?): User?
}